app/scripts/contentscript.js

```diff
@@ -64,18 +54,23 @@ let extensionMux,
 /**
  * Injects a script tag into the current document
  *
- * @param {string} content - Code to be executed in the current document
+ * @param {string} src - Path to code to be executed in the current document.
+ * @example injectScript(chrome.extension.getURL('inpage.js'))
  */
-function injectScript(content) {
+function injectScript(src) {
   try {
-    const container = document.head || document.documentElement;
-    const scriptTag = document.createElement('script');
-    scriptTag.setAttribute('async', 'false');
-    scriptTag.textContent = content;
-    container.insertBefore(scriptTag, container.children[0]);
-    container.removeChild(scriptTag);
+    const script = document.createElement("script");
+    // A script that has been injected into the DOM is executed asynchronously
+    // by default, but we need inpage.js to block so `window.ethereum` is
+    // available before the page's own scripts load.
+    script.async = false
```

Comment:
Can you provide a live demo where this works?
The internet says it should, but I can't seem to, which leaves me a bit skeptical regarding this technique..
Here's my demo https://weizmangal.com/public/demo-temp.html?silent&msg=111